core/src/main/scala/kafka/utils/timer/Timer.scala

```diff
@@ -43,7 +81,7 @@ class Timer(taskExecutor: ExecutorService, tickMs: Long = 1, wheelSize: Int = 20
   def add(timerTask: TimerTask): Unit = {
     readLock.lock()
     try {
-      addTimerTaskEntry(new TimerTaskEntry(timerTask))
+      addTimerTaskEntry(new TimerTaskEntry(timerTask, timerTask.delayMs + System.currentTimeMillis()))
```

Comment:
The main thing is that if you call that method multiple times it will return a different value each time (even if you don't do anything else). Such methods are considered `impure`, which is the term I prefer using because it's more useful and clearer than `side-effects`. Some may claim that `currentTimeMillis` is not side-effecting.